### DISCUSSION TITLE
About the link

### DISCUSSION THREAD

**Discussion by avgnerd** [spencerwooo/onedrive-vercel-index#127] [Q&A] [answered]
How do I create a 1drv link?

**Comment by spencerwooo** [comment, marked as answer]
Can you elaborate? Clicking on the copy button on the right side of the file in folder view (as shown below) ...

![image](https://user-images.githubusercontent.com/32114380/135061288-4b5b8322-2cda-4c9f-aa90-7327f6e0506c.png)

... or the copy direct link in file preview view (as shown below) ...

![image](https://user-images.githubusercontent.com/32114380/135061503-ba115052-486c-4882-8e43-c2d325ae6d4d.png)

both copies its `?raw` link which is then 302 redirected to its direct `1drv` link.

The download button, on the other hand, uses the `1drv` link.

It's worth noting that the `1drv` link you are looking for expires after 1 hour, so it's not a permanent link to the file. Hence we are copying the `?raw` file link which is generated by `onedrive-vercel-index`, and then lets it handle the redirect to the current available `1drv` link you are expecting.
